src/cryptography/hazmat/backends/openssl/x509.py

```diff
@@ -162,6 +162,40 @@ def public_bytes(self, encoding):
         self._backend.openssl_assert(res == 1)
         return self._backend._read_mem_bio(bio)
 
+    def is_issued_by(self, issuer_candidate):
+        if issuer_candidate.subject != self.issuer:
+            raise x509.base.InvalidIssuer(
+                expected=self.issuer, received=issuer_candidate.subject)
+
+        pkey = issuer_candidate.public_key()
+        signature = self.signature
+        data = self.tbs_certificate_bytes
+        try:
+            if isinstance(pkey, rsa.RSAPublicKeyWithSerialization):
+                pkey.verify(
+                    signature, data,
+                    padding=padding.PKCS1v15(),
+                    algorithm=self.signature_hash_algorithm,
+                )
+            elif isinstance(pkey, dsa.DSAPublicKeyWithSerialization):
+                pkey.verify(
+                    signature, data,
+                    algorithm=self.signature_hash_algorithm,
+                )
+            elif isinstance(pkey, ec.EllipticCurvePublicKeyWithSerialization):
+                # EC verify() requires instance of ec.ECDSA
+                pkey.verify(
+                    signature, data,
+                    signature_algorithm=ec.ECDSA(self.signature_hash_algorithm))
+            else:
+                raise UnsupportedAlgorithm(
+                    '{} verification is not implemented'
+                        .format(pkey.__class__)
+                )
+        except InvalidSignature:
```

Comment:
What is the point of this try-except?

Reply:
Good catch! This slipped through the self-review. I'm removing the redundant code and force-pushing the PR branch (for reference, leaving the original PR in branch [is_issued_by-v1](https://github.com/KonstantinShemyak/cryptography/tree/is_issued_by-v1) .